src/docker/registry.nim

```diff
@@ -369,12 +370,19 @@ iterator getConfigs(self: DockerImage, use: RegistryUse): RegistryConfig =
     if not isBuildx:
       buildx()
 
+var jsonCache = initTable[
+  (DockerImage, HttpMethod, string, RegistryUse),
+  (string, Response)
+]()
```

Comment:
to avoid global mutation, could this be its own module with set/get API to keep access to this state more uniform?

Reply:
🤷 for now its meant to be transparent on the caller when using registry SDK, stuff just gets cached transparently. we use this pattern in a bunch of places. eventually we can add util to manage cache however dont think it will win much in terms of consiseness